### DISCUSSION TITLE
Skeleton Rows - Lingering at the top of the table after sorting or filtering

### DISCUSSION THREAD

**Discussion by kscottthomas** [KevinVandy/material-react-table#583] [Q&A] [answered]
Not sure how to troubleshoot this, but when I sort or filter I get a few Skeleton rows left at the top of the table. As I sort and filter the number of Skeleton rows seems to increase.

![image](https://github.com/KevinVandy/material-react-table/assets/115663705/1602f361-61dc-45ad-83b6-efb07574996c)

Thank you in advance!
Scott

**Comment by KevinVandy** [comment]
Do you have some weird memoization going on? Never seen this.

**Reply by kscottthomas** [reply]
I may have found the issue... no memoization being used however I am seeing this in the console when it happens... It looks like my unique query has become non-unique. I'll post if fixing the query resolves the issue.

![image](https://github.com/KevinVandy/material-react-table/assets/115663705/e6b9f8bf-5e63-4cf3-81cf-a134d13717f9)

**Reply by KevinVandy** [reply]
Ah, so maybe you are using getRowId.

**Reply by kscottthomas** [reply, marked as answer]
Yes that is correct. The column that is considered to be the key and being set to the getRowID is being duplicated due to a poorly written SQL query. Thank you for getting to this so quickly!

**Reply by KevinVandy** [reply]
I ran into this myself. For me it was caused by setting the getRowId to fallback to an empty string instead of undefined. Which causes a type error. I'll have an internal fix for this soon so that while `state.isLoading` is true, getRowId will be off internally.
